odoo/tools/misc.py

```diff
@@ -1372,36 +1373,42 @@ def babel_locale_parse(lang_code):
         except:
             return babel.Locale.parse("en_US")
 
-def formatLang(env, value, digits=None, grouping=True, monetary=False, dp=False, currency_obj=False):
+def formatLang(env, value, digits=2, grouping=True, monetary=False, dp=None, currency_obj=None, rounding_mode=ROUND_HALF_EVEN, amount_rounding='decimals'):
     """
         Assuming 'Account' decimal.precision=3:
             formatLang(value) -> digits=2 (default)
             formatLang(value, digits=4) -> digits=4
             formatLang(value, dp='Account') -> digits=3
             formatLang(value, digits=5, dp='Account') -> digits=5
     """
+    # We don't want to return 0
+    if isinstance(value, str) and not value:
+        return ''
 
-    if digits is None:
-        digits = DEFAULT_DIGITS = 2
+    if amount_rounding == 'decimals':
         if dp:
-            decimal_precision_obj = env['decimal.precision']
-            digits = decimal_precision_obj.precision_get(dp)
+            digits = env['decimal.precision'].precision_get(dp)
         elif currency_obj:
             digits = currency_obj.decimal_places
+    else:
+        digits = 0
 
-    if isinstance(value, str) and not value:
-        return ''
-
-    lang_obj = get_lang(env)
+    if amount_rounding == 'thousands':
+        value = value / 10**3
+    elif amount_rounding == 'lakhs':
+        value = value / 10**5
+    elif amount_rounding == 'millions':
+        value = value / 10**6
```

Comment:
```suggestion
    rounding_mapping = {
        'decimals': 1,
        'thousands': 10**3,
        'lakhs': 10**5,
        'millions': 10**6,
    }
    value /= rounding_mapping.get(amount_rounding, 1)
```
I think using a dict would look better than a bunch of `if`/`elif` statements, but feel free to disagree :nerd_face:

Reply:
This is **way** better than what I had. Good catch !